src/umpire/strategy/ZeroByteHandler.cpp

```diff
@@ -0,0 +1,90 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/ZeroByteHandler.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#include "umpire/ResourceManager.hpp"
+
+namespace umpire {
+namespace strategy {
+
+ZeroByteHandler::ZeroByteHandler(
+  std::unique_ptr<AllocationStrategy>&& allocator) noexcept :
+AllocationStrategy(allocator->getName(), allocator->getId()),
+m_allocator(std::move(allocator)),
+m_zero_byte_pool(nullptr)
+{
+}
+
+void*
+ZeroByteHandler::allocate(std::size_t bytes)
+{
+  if (0 == bytes) {
+    UMPIRE_LOG(Debug, "Allocating 0 bytes for" << m_allocator->getName());
+    if (!m_zero_byte_pool)
```

Comment:
Move to constructor as well, if possible

Reply:
Impossible, at the moment :(

Reply:
Ah yes, ZeroByteHandler is created when ResourceManager is constructed.